apps/concierge_site/lib/controllers/auth_controller.ex

```diff
@@ -0,0 +1,105 @@
+defmodule ConciergeSite.AuthController do
+  use ConciergeSite.Web, :controller
+
+  require Logger
+
+  alias AlertProcessor.Model.User
+  alias AlertProcessor.Repo
+  alias ConciergeSite.SessionHelper
+  alias Plug.Conn
+
+  plug(Ueberauth)
+
+  @spec register(Conn.t(), map()) :: Conn.t()
+  def register(conn, _params) do
+    base_uri = System.get_env("KEYCLOAK_BASE_URI")
+    client_id = System.get_env("KEYCLOAK_CLIENT_ID")
+    redirect_uri = "KEYCLOAK_REDIRECT_URI" |> System.get_env() |> URI.encode()
```

Comment:
**suggestion (non-blocking):** it might be more Elixir-y to load these into application environment variables via `config/runtime.exs`.